06_modules_lab/03_bonus.py

```diff
@@ -1,13 +1,34 @@
-""" Write a program that searches current working directory
-for files larger than 1MB. Every time you find such a file print
-its name to the user.
-
-- When the program finds a large file. It should ask the user
-  a message asking if she wants to delete it, and delete the
-  file if requested
-
-- Take threshold and path as command line arguments
-
-Bonus: Use argparse module to parse command line arguments
-"""
-
+'''
+Searches in directory recursively for files larger than specific size and asks user if he wants to delete them.
+Receives the directory and minimum size of file as script parameters. Uses argparse module to read the parameters.
+Usage: python 03.py <directoryPath> <minimumFileSizeInKb>
+'''
+
+import sys
+import os
+import argparse
+
+parser = argparse.ArgumentParser(description='Find big files')
+parser.add_argument('directory', metavar='D', help="Root directory to search")
+parser.add_argument('minSizeInKb', metavar='S', type=int, help="Minimum file size in Kbytes")
+args = parser.parse_args()
+
+directory = args.directory
+minSizeInKb = args.minSizeInKb
+
+if not os.path.exists(directory):
+    print '%s does not exists' % directory
+    sys.exit(1)
+
+for root, dirs, files in os.walk(directory):
+    for file in files:
+        sizeInKb = os.stat(root + "\\" + file).st_size / 1024.0

```

Comment:
Python has `os.path.join` which joins multiple path components, so prefer `os.path.join(root, file)`. That would also work on unix where path separator is `/`